jax/_src/lax/lax.py

```diff
@@ -496,7 +496,13 @@ def concatenate(operands: Sequence[Array], dimension: int) -> Array:
 Precision = xla_client.PrecisionConfig.Precision
 Precision.__str__ = lambda precision: precision.name
 PrecisionType = Any
-PrecisionLike = Union[None, PrecisionType, Tuple[PrecisionType, PrecisionType]]
+PrecisionLike = Union[None, str, PrecisionType,
+                      Tuple[PrecisionType, PrecisionType]]
+_precision_strings = {
+    'bfloat16':      Precision.DEFAULT,
+    'tensorfloat32': Precision.HIGH,
```

Comment:
I'm wondering whether `tensorfloat32` is an good description. On A100 I believe CUDA `tensorfloat32` gives you  10 bits of mantissa on the input. And I'm not sure that `HIGH` in XLA's naming actually gives you `tensorfloat32` on GPU.

On TPU it means something different, namely a multipass algorithm on bfloat16 inputs. Is conflating the two is wise? I suspect we should just have two different names here.

Reply:
I guess the nice thing about strings instead of enums is that we can support multiple redundant names -- they don't have to be unique, and we can extend it over time.

So perhaps we could support all of `bfloat16`/`bfloat16_1x`, `bfloat16_3x` and `float32`/`bfloat16_6x` for now, and extend that list later when XLA adds true support for `tensorfloat32` on GPUs, in whatever form that takes.

Reply:
I suggest you want a couple of generic options:
* fastest
* most precise

and after that point you want to be completely specific about which algorithm you mean.

Reply:
OK, so for the current state of things, how about:

- `precision='highest'` -> `lax.Precision.HIGHEST`
- `precision='float32'` -> `lax.Precision.HIGHEST`
- `precision='bfloat16_3x'` -> `lax.Precision.HIGH`
- `precision='bfloat16'` -> `lax.Precision.DEFAULT`
- `precision='fastest'` -> `lax.Precision.DEFAULT`

(We could add the `bfloat16_1x` and `bfloat16_6x` aliases as well, if desired)

Reply:
cc @jekbradbury 

Maybe we should discuss on chat for lower latency?